icu4c/source/i18n/dtitvfmt.cpp

```diff
@@ -1023,29 +1026,23 @@ DateIntervalFormat::normalizeHourMetacharacters(const UnicodeString& skeleton) c
         }
         
         if (hourChar == CAP_H || hourChar == LOW_K) {
-            result.replace(metacharStart, metacharCount, hourChar);
+            result.replace(hourFieldStart, hourFieldLength, hourChar);
```

Comment:
OK, so I get that the above replace does not change the length of result, so the dayPeriodStart index is still valid for the remove below. But later on...

Reply:
Ugh.  You're absolutely right.  I'm an idiot.  Give me a little while, and I'll put up a new commit that addresses that.